src/rewriting/rewriting_rule.rs

```diff
@@ -1256,15 +1244,21 @@ impl<'a> RewriteVisitor<'a> for Rewriter<'a> {
                     [Property::PrivacyUnitPreserving, Property::Published],
                     Property::PrivacyUnitPreserving,
                     Parameters::PrivacyUnit(privacy_unit),
-                ) | (
+                )
+                | (
                     [Property::PrivacyUnitPreserving, Property::DifferentiallyPrivate],
                     Property::PrivacyUnitPreserving,
                     Parameters::PrivacyUnit(privacy_unit),
+                )
+                | (
+                    [Property::PrivacyUnitPreserving, Property::Public],
+                    Property::PrivacyUnitPreserving,
+                    Parameters::PrivacyUnit(privacy_unit),
                 ) => {
                     let privacy_unit_tracking = PrivacyUnitTracking::new(
-                        self.0,
+                        self.relations,
                         privacy_unit.clone(),
-                        crate::privacy_unit_tracking::Strategy::Hard,
+                        self.strategy,
```

Comment:
This will fail when not hard. You should not change that. Change the rule setter instead.